test/indexer/test_hca_indexer.py

```diff
@@ -1276,6 +1276,89 @@ def test_multiple_samples(self):
+        """
+        # Sample (Specimen): 70d2b85a, Donor: b111e5bf
+        bundle_fqid = self.bundle_fqid(uuid='1fd499c5-f397-4bff-9af0-eb42c37d5fbe',
+                                       version='2021-03-18T11:38:49.884Z')
+        self._index_canned_bundle(bundle_fqid)
+        # Sample (Organoid): df23c109, Donor: none
+        bundle_fqid = self.bundle_fqid(uuid='0722b70c-6778-423d-8fe9-869e2a515d35',
+                                       version='2021-03-18T11:38:49.863Z')
+        self._index_canned_bundle(bundle_fqid)
+        donor_b111e5bf = {
+            'document_id': 'b111e5bf-e907-47f9-8eed-75b2ec5536c5',
+            'biomaterial_id': 'Human_62',
+            'biological_sex': 'male',
+            'genus_species': ['Homo sapiens'],
+            'development_stage': 'human adult stage',
+            'diseases': ['normal'],
+            'organism_age': '62 year',
+            'organism_age_value': '62',
+            'organism_age_unit': 'year',
+            'organism_age_range': {
+                'gte': 1955232000.0,
+                'lte': 1955232000.0
+            }
+        }
+        donor_none = {
+            'document_id': None,
+            'biomaterial_id': None,
+            'biological_sex': None,
+            'genus_species': [None],
+            'development_stage': None,
+            'diseases': [None],
+            'organism_age': None,
+            'organism_age_value': None,
+            'organism_age_unit': None
+        }
+        hits = self._get_all_hits()
+        for hit in hits:
+            contents = hit['_source']['contents']
+            entity_type, aggregate = self._parse_index_name(hit)
+            if entity_type == 'projects':
+                if aggregate:
+                    expected = dict(donor_b111e5bf)
+                    # FIXME: The donor inner entity of a project aggregate should
+                    #        include `None` in each of the aggregated donor fields
+                    #        since at least one of the bundles has no donor entity.
+                    #        https://github.com/databiosphere/azul/issues/3152
+                    if False:
+                        for key, val in expected.items():
+                            if key == 'organism_age_range':
+                                expected[key] = [val]
+                            elif isinstance(val, list):
+                                expected[key] += donor_none[key]
+                            else:
+                                expected[key] = [val, donor_none[key]]
+                    else:
+                        for key, val in expected.items():
+                            if not isinstance(val, list):
+                                expected[key] = [val]
+                    expected.update(
+                        {
+                            'donor_count': 1,
+                            'donor_count_': 1
+                        }
+                    )
```

Comment:
It's confusing and inefficient that this is updating the expectation for every hit, in exactly the same way.